### PR TITLE
Add support for VPCOptions in ElasticSearch

### PR DESCRIPTION
**Pull request by Hyunk3l**
Hello :)

I'm trying to solve this issue I opened some minutes ago https://github.com/cloudtools/troposphere/issues/861

The idea is to add support for VPCOptions in ElasticSearch (you will find more info in the issue).

Hope this works!

### CODE REVIEW

**Comment by Hyunk3l** [comment]
As @craigbruce pointed out, seems that CloudFormation does not support VCPOptions https://github.com/cloudtools/troposphere/issues/861#issuecomment-338049650
Should I close this PR or wait for them to implement the option?

**Comment by markpeek** [comment]
I created a new label for it so we can leave it here in case it gets implemented soon in CF. Thanks.

**Comment by markpeek** [comment]
@Hyunk3l I see this got implemented although it isn't a dict but rather a specific property type. Docs are [here](http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-elasticsearch-domain-vpcoptions.html). Can you update this PR? Thanks.

File: troposphere/elasticsearch.py [hunk 1]
```diff
@@ -66,9 +66,15 @@ class Domain(AWSObject):
         'ElasticsearchClusterConfig': (ElasticsearchClusterConfig, False),
         'ElasticsearchVersion': (basestring, False),
         'SnapshotOptions': (SnapshotOptions, False),
-        'Tags': ((Tags, list), False)
+        'Tags': ((Tags, list), False),
+        'VPCOptions': (VPCOptions, False)
     }
 
+class VPCOptions(AWSProperty):
```
Comment (Hyunk3l) [inline troposphere/elasticsearch.py]: @markpeek `AWSProperty`, right?

File: troposphere/elasticsearch.py [hunk 2]
```diff
@@ -55,6 +55,13 @@ class SnapshotOptions(AWSProperty):
     }
 
 
+class VPCOptions(AWSProperty):
+    props = {
+        "SecurityGroupIds": (dict, False),
```
Comment (markpeek) [inline troposphere/elasticsearch.py]: These should be list of strings:
    "SecurityGroupIds": ([basestring], False),
    "SubnetIds": ([basestring], False)

File: examples/ElasticsearchDomain.py [hunk 3]
```diff
@@ -36,7 +36,14 @@
                         'Action': 'es:*',
                         'Resource': '*'
                     }]},
-    AdvancedOptions={"rest.action.multi.allow_explicit_index": "true"}
+    AdvancedOptions={"rest.action.multi.allow_explicit_index": "true"},
+    VPCOptions={'SubnetIds': [
```
Comment (markpeek) [inline examples/ElasticsearchDomain.py]: This should be (and you'll need an import at the top for VPCOptions):
VPCOptions=VPCOptions(
    SubnetIds=["subnet-4f2bb123"],
    SecurityGroupIds=["sg-04cf048c"],
),

File: tests/examples_output/ElasticsearchDomain.template [hunk 4]
```diff
@@ -36,6 +36,14 @@
                 },
                 "SnapshotOptions": {
                     "AutomatedSnapshotStartHour": 0
+                },
+                "VPCOptions": {
```
Comment (markpeek) [inline tests/examples_output/ElasticsearchDomain.template]: Fix this up given the change above.

File: examples/ElasticsearchDomain.py (hunk 3 above)
Comment (markpeek) [reply examples/ElasticsearchDomain.py]: Also, the "Ref" is a bit of a misnomer given the subnets/sg's aren't defined in this example. Really you're accessing already defined subnets/security groups.

File: troposphere/elasticsearch.py (hunk 2 above)
Comment (Hyunk3l) [reply troposphere/elasticsearch.py]: You are totally right!

**Comment by Hyunk3l** [comment]
Thanks @markpeek! All changes are done and tests are green :)

**Comment by markpeek** [comment]
Nice! Thank you for implementing this addition.

**Comment by Hyunk3l** [comment]
Thank you for your support! 😄

**Pull request merged**